Doppler.HtmlEditorApi/Repositories.DopplerDb/DopplerCampaignContentRepository.cs

```diff
@@ -124,17 +119,54 @@ public async Task SaveCampaignContent(string accountName, ContentData contentRow
             },
             // TODO: test this scenario
             // Probably a unit test will be necessary
-            _ => throw new NotImplementedException($"Unsupported campaign content type {contentRow.GetType()}")
+            _ => throw new NotImplementedException($"Unsupported campaign content type {content.GetType()}")
         };
 
-        IExecutableDbQuery upsertContentQuery = campaignStatus.ContentExists
-            ? new UpdateCampaignContentDbQuery(
+        await _dbContext.ExecuteAsync(
+            new InsertCampaignContentDbQuery
+            (
                 IdCampaign: queryParams.IdCampaign,
                 EditorType: queryParams.EditorType,
                 Content: queryParams.Content,
                 Head: queryParams.Head,
-                Meta: queryParams.Meta)
-            : new InsertCampaignContentDbQuery(
+                Meta: queryParams.Meta
+                ));
+
+        var updateCampaignStatusQuery = new UpdateCampaignStatusDbQuery(
+            setCurrentStep: 2,
+            setHtmlSourceType: UpdateCampaignStatusDbQuery.TEMPLATE_HTML_SOURCE_TYPE,
+            whenIdCampaignIs: content.campaignId,
+            whenCurrentStepIs: 1
+        );
+
+        await _dbContext.ExecuteAsync(updateCampaignStatusQuery);
+    }
+    public async Task UpdateCampaignContent(string accounName, ContentData content)
+    {
+        var queryParams = content switch
+        {
+            UnlayerContentData unlayerContentData => new
+            {
+                IdCampaign = unlayerContentData.campaignId,
+                Content = unlayerContentData.htmlContent,
+                Head = unlayerContentData.htmlHead,
+                Meta = unlayerContentData.meta,
+                EditorType = (int?)EDITOR_TYPE_UNLAYER
+            },
+            HtmlContentData htmlContentData => new
+            {
+                IdCampaign = htmlContentData.campaignId,
+                Content = htmlContentData.htmlContent,
+                Head = htmlContentData.htmlHead,
+                Meta = (string)null,
+                EditorType = (int?)null
+            },
```

Comment:
I think that logic would be refactored in a one method reusable.

Reply:
Creo que la refactorización no es necesaria, simplemente queremos mapear el `content` en los parámetros de SQL.

Antes usabamos un objeto anónimo porque había que revisar otra condición más (si era insert o update), eso ahora ya no es necesario y podemos simplificarlo así:

```suggestion
        var updateContentQuery = content switch
        {
            UnlayerContentData unlayerContentData => new UpdateCampaignContentDbQuery(
                IdCampaign: unlayerContentData.campaignId,
                Content: unlayerContentData.htmlContent,
                Head: unlayerContentData.htmlHead,
                Meta: unlayerContentData.meta,
                EditorType: (int?)EDITOR_TYPE_UNLAYER
            ),
            HtmlContentData htmlContentData => new UpdateCampaignContentDbQuery(
                IdCampaign: unlayerContentData.campaignId,
                Content: unlayerContentData.htmlContent,
                Head: unlayerContentData.htmlHead,
                Meta: null,
                EditorType: null
            },
```